### PR TITLE
Front end testing

### LINKED ISSUE #54: Add frontend tests
## User Story
As a developer, I would like to have a tested frontend codebase. so that it is less prone to bugs.

## Detailed Description

We need to decide if we are going to do this automated or manual.

### Automated approach
Add frontend tests using jest and enzyme. Check out this tutorial: https://blog.sapegin.me/all/react-testing-1-best-practices/ 

### Manual approach

Create a frontend testing checklist as a markdown file.

We have to plan the tests anyway, so plan it out on paper and then one of the following:
- write automated tests
- expand out the paper plan

## Subtasks (Please also link task PRs here)
- []
- []
- []

### PR DESCRIPTION
**Pull request by akanksha003**
Closes #54 .

Frontend Testing using Jest and Enzyme. Most important components to test are Chart, Watchlist, Portfolio, BuyAndSellWidget, and Login.

Login and Portfolio tests are complete.

### CODE REVIEW

**Comment by akanksha003** [comment]
Reminder to self: remove console log statements lol

File: client/src/pages/chart.test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                price_per_share: "100",
+                date_changed: "2021-03-02T01:22:47.086Z"
+            }
+        ]
+    }
+})
+
+
+describe('ChartComponent tests', () => {
+    require('../services/portfolioService');
```
Comment (willyspinner) [inline client/src/pages/chart.test.js:39]: If this whole suite is blank, can we comment it out?

**Review by willyspinner — CHANGES_REQUESTED** [review]
Nice!!! Thank you so much for doing this. Can we also make sure that in `.travis.yml`, we delete the `--passWithNoTests` option in the command to run the frontend tests? 

This is because we now have frontend tests :)

File: client/src/pages/chart.test.js (hunk 1 above)
Comment (akanksha003) [reply client/src/pages/chart.test.js:39]: hey so if commenting that out would make the test fail :( 
We could instead delete the chart.test.js file, but I wanted to keep it for later use, which is why I left some minimal code that wouldn't cause the test to fail :D

Comment (willyspinner) [reply client/src/pages/chart.test.js:39]: Ahh gotcha, that's okay then!

**Comment by akanksha003** [comment]
Ok! Made the requested changes. Waiting for checks to complete :)

**Review by willyspinner — APPROVED** [review]
ready to go! Go ahead and merge

**Pull request merged**